ide/parsing.api/src/org/netbeans/modules/parsing/api/Source.java

```diff
@@ -334,81 +332,72 @@ public Snapshot createSnapshot () {
+                                        // CR-LF pair changed to single LF
+                                        continue;
                                     }
-
-                                    int [] lsoArr = new int [lso.size()];
-                                    int idx = 0;
-                                    for(Integer offset : lso) {
-                                        lsoArr[idx++] = offset;
+                                    if (ch == CR) {
+                                        // convert to LF; subsequent LF will be skipped
+                                        output.append(LF);
+                                        lso.add(output.length());
+                                        lastCharCR = true;
+                                    } else if (ch == LS || ch == PS) { // Unicode LS, PS
+                                        output.append(LF);
+                                        lso.add(output.length());
+                                        lastCharCR = false;
+                                    } else { // current char not CR
+                                        lastCharCR = false;
+                                        output.append(ch);
                                     }
-
-                                    text[0] = output;
-                                    lineStartOffsets[0] = lsoArr;
-                                } finally {
-                                    reader.close ();
                                 }
-                            } finally {
-                                is.close ();
                             }
-                        } else {
-                            LOG.log(
-                                Level.WARNING,
-                                "Source {0} of size: {1} has been ignored due to large size. Files large then {2} bytes are ignored, you can increase the size by parse.max.file.size property.",  //NOI18N
-                                new Object[]{
-                                    FileUtil.getFileDisplayName(fileObject),
-                                    fileObject.getSize(),
-                                    Utilities.getMaxFileSize()
-                                });
+
+                            int[] lsoArr = new int[lso.size()];
+                            int idx = 0;
+                            for(Integer offset : lso) {
+                                lsoArr[idx++] = offset;
+                            }
+
+                            text[0] = output;
+                            lineStartOffsets[0] = lsoArr;
+                        } catch (FileNotFoundException fnfe) {
+                            // working with a stale FileObject, just ignore this (eg see #158119)
+                        } catch (CoderMalfunctionError cme) {
+                            text[0] = ""; //NOI18N
+                            lineStartOffsets[0] = new int[] { 0 };
```

Comment:
Question/nitpick: is there a reason to treat `CoderMalfunctionError` different compared to `IOException` with regard to the variables? I see a reason for the different logging, but why handle the variable differently?

Reply:
yeah I suppose we can handle both the same way. I took the two lines from the OutOfMemory catch block - so I guess all paths which are not able to complete normally (like out of memory) should be the same handler with a different message.

will take another look

Reply:
done, used multicatch - FileNotFound is also implicitly handled via IOException.